### PR TITLE
Dev/add entry page

### PR DESCRIPTION
**Pull request by binnmti**
Create Entry Page that anyone can see
image

## Summary by CodeRabbit

- **New Features**
	- Introduced new routes for the Entry and Document pages, enabling users to access specific entries and documents using unique identifiers.
	- Added the Entry component to display document content dynamically, with error handling for permissions and missing data.
	- Introduced the DocumentPage component for viewing documents based on URL parameters.
	- Added the MarkdownRenderer component to enhance the rendering of Markdown content within the application.
	- Enhanced the Edit component with updated Markdown rendering and a new "Published" button for saving documents.

- **Bug Fixes**
	- Implemented error messaging for routing issues and access restrictions, enhancing the user navigation experience.

### CODE REVIEW

File: one-pager-maker/src/pages/Entry.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    // this is called when route setting is wrong
+    return <main>Route setting is wrong</main>;
+  }
+  const documentResult = documentApi.useGetDocumentQuery({ uid, documentId });
+  const document = documentResult.data?.value;
+  console.log("document", document);
+  if (
+    documentResult.data?.error &&
+    documentResult.data?.error.code === "permission-denied"
+  ) {
```
[the remark this answers is not in the document]
Comment (binnmti) [reply one-pager-maker/src/pages/Entry.tsx]: This code has been deleted

File: one-pager-maker/src/pages/Entry.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  const { uid, documentId } = useParams<{
+    uid: string;
+    documentId: string;
+  }>();
+  if (uid === undefined || documentId === undefined) {
+    return <main>Route setting is wrong</main>;
+  }
+  const documentResult = documentApi.useGetDocumentQuery({ uid, documentId });
+  const document = documentResult.data?.value;
+  return (
```
Comment (Harineko0) [inline one-pager-maker/src/pages/Entry.tsx:15]: nits: linebreak before return statement

File: one-pager-maker/src/pages/Entry.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    <>
+      <div>
```
Comment (Harineko0) [inline one-pager-maker/src/pages/Entry.tsx:17]: nits: These `<>` and `<div>` might be unnecessary?\

File: one-pager-maker/src/pages/Entry.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        <div className="flex h-svh w-full p-1">
+          <div className="w-full overflow-visible overflow-scroll overflow-x-hidden border p-1">
+            <Markdown className="markdown">{document?.contents}</Markdown>
```
Comment (Harineko0) [inline one-pager-maker/src/pages/Entry.tsx:20]: To follow DRY principle, Markdown component should be declared in other file like below:
```
return (
  <Markdown className={"markdown"}>{children}</Markdown>
);
```

File: one-pager-maker/src/Router.tsx [hunk 5]
```diff
@@ -30,6 +31,7 @@ export const Router = () => {
           />
         }
       />
+      <Route path="/entry/:uid/:documentId" element={<Entry />} />
```
[the remark this answers is not in the document]
Comment (binnmti) [reply one-pager-maker/src/Router.tsx:34]: # qitta
- /drafts/itemId/edit
- /userId/items/itemId

# haetna
- /userid/edit?entry=entryId
- /userid/entry/datetime/entryId

# zenn
- /articles/articlesId/edit
- /userId/articles/articlesId

[the remark this answers is not in the document]
Comment (binnmti) [reply one-pager-maker/src/Router.tsx:34]: For now, I matched it with edit.  I thought the One Pager maker document was not an article.
image

So I'll do the following
```/:uid/document/:documentId```

[the remark this answers is not in the document]
Comment (Harineko0) [reply one-pager-maker/src/Router.tsx:34]: I suggest usIng `documents` rather than `document`

[the remark this answers is not in the document]
Comment (binnmti) [reply one-pager-maker/src/Router.tsx]: Changed to documents

File: one-pager-maker/src/pages/Entry.tsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
+import Markdown from "react-markdown";
+import { useParams } from "react-router-dom";
+import { documentApi } from "../api/documentApi.ts";
+
+function Entry() {
```
[the remark this answers is not in the document]
Comment (binnmti) [reply one-pager-maker/src/pages/Entry.tsx]: Add `Page`

File: one-pager-maker/src/pages/Entry.tsx [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  const { uid, documentId } = useParams<{
+    uid: string;
+    documentId: string;
+  }>();
+  if (uid === undefined || documentId === undefined) {
```
[the remark this answers is not in the document]
Comment (binnmti) [reply one-pager-maker/src/pages/Entry.tsx]: I don't really have any rules when it comes to line breaks, but do you have any?

File: one-pager-maker/src/pages/Entry.tsx (hunk 3 above)
Comment (binnmti) [reply one-pager-maker/src/pages/Entry.tsx]: delete `<>`

File: one-pager-maker/src/pages/Entry.tsx (hunk 4 above)
Comment (binnmti) [reply one-pager-maker/src/pages/Entry.tsx]: It's never DRY

File: one-pager-maker/src/pages/DocumentPage.tsx [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+  const documentResult = documentApi.useGetDocumentQuery({ uid, documentId });
+  const document = documentResult.data?.value;
+
+  return (
+    <div>
+      <div className="flex h-svh w-full p-1">
+        <div className="w-full overflow-visible overflow-scroll overflow-x-hidden border p-1">
+          <MarkdownRenderer
+            contents={document?.contents || ""}
+          ></MarkdownRenderer>
```
Comment (Harineko0) [inline one-pager-maker/src/pages/DocumentPage.tsx:23]: This end tag seems unnecessary for me.
Is is okay to write like `<MarkdownRenderer contents={document?.contents || ""} />` instead?

File: one-pager-maker/src/pages/Edit.tsx [hunk 9]
```diff
@@ -136,6 +132,13 @@ function Edit() {
           >
             Save
           </button>
+          <button
+            className="border-4 border-solid"
+            type="button"
+            onClick={() => onClickSave()}
```
Comment (Harineko0) [inline one-pager-maker/src/pages/Edit.tsx:138]: I suppose it is no problem to pass function directly like below:
```
onClick={onClickSave}
```

**Review by Harineko0 — COMMENTED** [review]
LGTM!

File: one-pager-maker/src/pages/Entry.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (Harineko0) [reply one-pager-maker/src/pages/Entry.tsx]: Sumire san said like that

**Review by Harineko0 — APPROVED** [review]
Approved b

**Pull request merged**